borsh-schema-derive-internal/src/attribute_helpers/field.rs

```diff
@@ -0,0 +1,191 @@
+#![allow(unused)]
+// TODO: remove unused when unsplit is done
+use std::collections::BTreeMap;
+
+use once_cell::sync::Lazy;
+use syn::{meta::ParseNestedMeta, Attribute, ExprPath, WherePredicate};
+
+use self::{bounds::BOUNDS_FIELD_PARSE_MAP, schema::SCHEMA_FIELD_PARSE_MAP};
+
+use super::{
+    parsing::{attr_get_by_symbol_keys, meta_get_by_symbol_keys, parse_lit_into},
+    BoundType, Symbol, BORSH, BOUND, DESERIALIZE_WITH, PARAMS, SCHEMA, SERIALIZE_WITH, SKIP,
+    WITH_FUNCS,
+};
+
+pub mod bounds;
+pub mod schema;
+
+enum Variants {
+    Schema(schema::Attributes),
+    Bounds(bounds::Bounds),
+    SerializeWith(syn::ExprPath),
+    DeserializeWith(syn::ExprPath),
+}
+
+type ParseFn = dyn Fn(Symbol, Symbol, &ParseNestedMeta) -> syn::Result<Variants> + Send + Sync;
+
+static BORSH_FIELD_PARSE_MAP: Lazy<BTreeMap<Symbol, Box<ParseFn>>> = Lazy::new(|| {
```

Comment:
Is this pattern inherited from serde? I feel it is a bit hard to orient myself with these indirections and passing parsing functions as a map.

Reply:
Code in this pr, duplicates `From` implementation blocks (like `impl From<BTreeMap<Symbol, Variants>> for Bounds`) for each group of  [meta item](https://doc.rust-lang.org/reference/attributes.html#meta-item-attribute-syntax)-s,  found at the same level of nesting in topmost meta item.
Serde's approach is to duplicate funcs similar to `fn attr_get_by_symbol_keys` for each such group of meta items, making small adjustments for each case.
This pr differs from serde that it duplicates simpler code than the one which serde duplicates, but makes `get_nested_meta_logic` a little bit more complex, than in serde, to make it more generic.